frontends/PyCDE/src/signals.py

```diff
@@ -137,7 +137,7 @@ def name(self):
     from .circt.dialects import hw
     if isinstance(owner, ir.Block) and isinstance(owner.owner, hw.HWModuleOp):
       block_arg = ir.BlockArgument(self.value)
-      mod_type = hw.ModuleType(owner.owner.module_type.value)
+      mod_type = hw.ModuleType(ir.TypeAttr(owner.owner.module_type).value)
```

Comment:
Just FYI- these explicit casts were necessitated from the most recent LLVM bump which changed some things related to the automatic casting of builtin types and attributes. In the latest LLVM, you can enable the automatic casting by depending on the builtin dialect Python files like I mentioned here: https://github.com/llvm/circt/pull/6404#issuecomment-1811761298

There is still some discussion about how automatic casting for builtin dialect should work upstream: https://github.com/llvm/llvm-project/pull/72338. In a future LLVM bump, we might be able to get the automatic casting behavior for free again, or we might decide upstream that we need to include the builtin dialect's Python files like we've done for the CIRCTPythonModules target in lib/Bindings/Python.

Reply:
Yeah, I saw that comment awhile back (but couldn't remember where it was). We do this sort of downcasting all over the place (when the type is well known), so it was easier/most consistent to just do it this way. I haven't been following the upstream discussion, but would rather wait for it to be settled then use it appropriately.